### PR TITLE
Fix line endings of `huffman-text.input` and re-enable test on Windows

### LINKED ISSUE #13892: std lib compression test is failing on windows: writeBlockHuff
### Zig Version

0.11.0-dev.730+cd9af0f28

### Steps to Reproduce and Observed Behavior

```
2022-12-11T09:26:51.4881378Z 1527/2081 test.writeBlockHuff... FAIL (TestUnexpectedResult)
2022-12-11T09:26:51.4883640Z D:\a\zig\zig\lib\std\testing.zig:509:14: 0x7ff7b4c02687 in expect (test.exe.obj)
2022-12-11T09:26:51.4886076Z     if (!ok) return error.TestUnexpectedResult;
2022-12-11T09:26:51.4886391Z              ^
2022-12-11T09:26:51.4930922Z D:\a\zig\zig\lib\std\compress\deflate\huffman_bit_writer.zig:912:5: 0x7ff7b546ce5e in testBlockHuff (test.exe.obj)
2022-12-11T09:26:51.4934293Z     try expect(mem.eql(u8, buf.items, want));
2022-12-11T09:26:51.4934588Z     ^
2022-12-11T09:26:51.4979739Z D:\a\zig\zig\lib\std\compress\deflate\huffman_bit_writer.zig:875:5: 0x7ff7b546d5b4 in test.writeBlockHuff (test.exe.obj)
2022-12-11T09:26:51.4982426Z     try testBlockHuff(
2022-12-11T09:26:51.4982719Z     ^
2022-12-11T09:27:01.5935445Z 1995 passed; 83 skipped; 3 failed.
2022-12-11T09:27:01.6015815Z error: the following test command failed with exit code 1:
2022-12-11T09:27:01.6016504Z D:\a\zig\zig\zig-cache\o\df5fd2aaea735523e9419229dce9f3fb\test.exe
2022-12-11T09:27:01.6162616Z error: test...
2022-12-11T09:27:01.6162995Z error: The following command exited with error code 1:
2022-12-11T09:27:01.6164731Z D:\a\zig\zig\build-release\stage3-release\bin\zig.exe test D:\a\zig\zig\lib\std\std.zig --test-name-prefix std-native-Debug-bare-multi-default  --cache-dir D:\a\zig\zig\zig-cache --global-cache-dir C:\Users\runneradmin\AppData\Local\zig --name test -fno-single-threaded -I D:\a\zig\zig\test -L D:\a\zig\zig\zig+llvm+lld+clang-x86_64-windows-gnu-0.11.0-dev.448+e6e459e9e\lib -I D:\a\zig\zig\zig+llvm+lld+clang-x86_64-windows-gnu-0.11.0-dev.448+e6e459e9e\include --zig-lib-dir D:\a\zig\zig\lib --enable-cache 
```

It's the standard library tests on x86_64-windows after removing this special case for Windows:

https://github.com/ziglang/zig/blob/cd9af0f286725c19f5dc0335fb1db1ab8cf7af0d/lib/std/std.zig#L103-L131

### Expected Behavior

All standard library tests pass, same as the other operating systems.

### PR DESCRIPTION
**Pull request by squeek502**
Closes #13892

### CODE REVIEW

**Comment by Vexu** [comment]
Is there anything preventing us from marking all files `eol=lf` to avoid issues like this in the future?

Also the attributes for the `deps/` folder should be removed since it no longer exists.

**Comment by andrewrk** [comment]
Shoutouts to 2c8495b4bb261d440bc6d1a6b0415a64c8d99222, a commit that took me several hours to debug why glibc stubs weren't being generated correctly on Windows.


> Is there anything preventing us from marking all files `eol=lf` to avoid issues like this in the future?

Yes, unfortunately if we do this, then it could cause the problem the *other* way. For example, perhaps there was a file that represented a raw HTTP header dump. It would incorrectly change the \r\n endings into \n, breaking things once more.

What we really want is `eol=nomangle`, if such an option existed. Telling Git to do what should really be the default, which is to not go around editing our source files without our permission!!

**Comment by squeek502** [comment]
Seems like the aarch64-windows CI still checked out `huffman-text.input` with CRLF line endings. Unsure what to do to get it to use the correct line endings. Is it possible that it's re-using a existing clone of the repo? If so, then this particular PR will need to force a fresh checkout of at least that particular file somehow.

**Comment by andrewrk** [comment]
> Is it possible that it's re-using a existing clone of the repo? If so, then this particular PR will need to force a fresh checkout of at least that particular file somehow.

yes, that is what is happening. understood, will look into it.

**Comment by squeek502** [comment]
I think if `git checkout-index --force -all` is run after the fetch, it would get the correct line endings.

File: .gitattributes [hunk 1]
```diff
@@ -2,6 +2,7 @@
 *.txt text eol=lf
 langref.html.in text eol=lf
 deps/SoftFloat-3e/*.txt text eol=crlf
+huffman-text.input text eol=lf
```
Comment (andrewrk) [inline .gitattributes:5]: How about this instead?

```suggestion
lib/std/compress/deflate/testdata/* binary
```

Comment (squeek502) [reply .gitattributes:5]: Will try that locally and verify that it'll work

Comment (squeek502) [reply .gitattributes]: Confirmed to work, might still have an issue with the CI not re-checking-out the relevant file though :(

**Comment by andrewrk** [comment]
I opened a new pull request to try to solve this with a couple extra commands in the CI script, which can be reverted in a later commit: #13999

**Pull request closed without merging**